### DISCUSSION TITLE
emission detail map for emission

### DISCUSSION THREAD

**Discussion by mddm1** [godotengine/godot-proposals#10378] [3D]
### Describe the project you are working on

a 3d game

### Describe the problem or limitation you are having in your project

I'm trying to make a flashlight with a battery UI that uses emission. 
![g](https://github.com/user-attachments/assets/11881a30-b9bb-4ea7-ad73-ada37094a2be)
 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

add another property called "emission detail" below "detail" that would affect the emission property

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

add another property called "emission detail" below "detail" that would affect the emission property

### If this enhancement will not be used often, can it be worked around with a few lines of script?

don't know

### Is there a reason why this should be core and not an add-on in the asset library?

it must be core

**Comment by AThousandShips** [comment]
Converted to a discussion as there's no details for how this might be implemented

**Comment by AThousandShips** [comment]
Note that this should be entirely possible to achieve using a custom shader, as the standard material isn't special for the rendering backend, it just creates a shader from the configuration, so this should be possible (if not trivial) to achieve by converting your material to a shader material and adding the required code, you can even still configure it freely

**Comment by Calinou** [comment]
- Related to https://github.com/godotengine/godot-proposals/issues/3499.

This should be relatively easy to add, except the emission operator used would likely have to be the same as for the main emission texture (if one is defined).

However, there isn't a lot of interest in adding more detail mapping-related features to BaseMaterial3D, so this may not be a good fit for core and may be better suited to custom shaders.

**Comment by mddm1** [comment]
understood. what you say is fair. I also ended up writing a custom shader material which works almost as intended. would had been a quality of life improvement to just have that out of the box. didn't know it is that difficult to implement/ compatibility wise. For me it was surprising there was a detail map for albedo but none for the emission. Or maybe all properties should have a detail map available. Is there a specific reason why only the Albedo has it? Also, aren't those 2 (albedo/emission) supposed to work the same way (regarding the implementation details) ?

**Reply by Calinou** [reply]
> Is there a specific reason why only the Albedo has it?

It was most likely implemented this way to prevent the shader from being too complex (and slow to render), while avoiding bloat in the inspector.
